### PR TITLE
add script and doc for cherry_pick_pull

### PR DESCRIPTION
**Pull request by carlory**
**What type of PR is this?**

/kind documentation
/kind feature

**What this PR does / why we need it**:

As [release-1.0](https://github.com/karmada-io/karmada/tree/release-1.0) published, we need a script to backport some PRs from master to release branches.

**Does this PR introduce a user-facing change?**:

```release-note
add a script and document for cherry-pick pull
```

### CODE REVIEW

**Comment by carlory** [comment]
```shell
➜  karmada git:(script-cherry-pick-pull) DRY_RUN=true ./hack/cherry_pick_pull.sh upstream/release-1.0 1206
You are not logged into any GitHub hosts. Run gh auth login to authenticate.

➜  karmada git:(script-cherry-pick-pull) gh auth login
? What account do you want to log into? GitHub.com
? What is your preferred protocol for Git operations? HTTPS
? Authenticate Git with your GitHub credentials? Yes
? How would you like to authenticate GitHub CLI? Paste an authentication token
Tip: you can generate a Personal Access Token here https://github.com/settings/tokens
The minimum required scopes are 'repo', 'read:org', 'workflow'.
? Paste your authentication token: ****************************************
- gh config set -h github.com git_protocol https
✓ Configured git protocol
✓ Logged in as carlory

➜  karmada git:(script-cherry-pick-pull) ./hack/cherry_pick_pull.sh upstream/release-1.0 1206
github.com
  ✓ Logged in to github.com as carlory (/Users/dijun/.config/gh/hosts.yml)
  ✓ Git operations for github.com configured to use https protocol.
  ✓ Token: *******************

+++ Updating remotes...
Fetching upstream
Fetching origin
+++ Creating local branch automated-cherry-pick-of-#1206-upstream-release-1.0-1641377057
Branch 'automated-cherry-pick-of-#1206-upstream-release-1.0-1641377057' set up to track remote branch 'release-1.0' from 'upstream'.
Switched to a new branch 'automated-cherry-pick-of-#1206-upstream-release-1.0-1641377057'
+++ Downloading patch to /tmp/1206.patch (in case you need to do this again)

+++ About to attempt cherry pick of PR. To reattempt:
  $ git am -3 /tmp/1206.patch

Applying: remove unreachable code

+++ I'm about to do the following to push to GitHub (and I'm assuming origin is your personal fork):

  git push origin automated-cherry-pick-of-#1206-upstream-release-1.0-1641377057:automated-cherry-pick-of-#1206-upstream-release-1.0

+++ Proceed (anything but 'y' aborts the cherry-pick)? [y/n] y
Enumerating objects: 11, done.
Counting objects: 100% (11/11), done.
Delta compression using up to 4 threads
Compressing objects: 100% (5/5), done.
Writing objects: 100% (6/6), 515 bytes | 515.00 KiB/s, done.
Total 6 (delta 4), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
remote:
remote: Create a pull request for 'automated-cherry-pick-of-#1206-upstream-release-1.0' on GitHub by visiting:
remote:      https://github.com/carlory/karmada/pull/new/automated-cherry-pick-of-%231206-upstream-release-1.0
remote:
To https://github.com/carlory/karmada.git
 * [new branch]        automated-cherry-pick-of-#1206-upstream-release-1.0-1641377057 -> automated-cherry-pick-of-#1206-upstream-release-1.0

+++ Creating a pull request on GitHub at carlory:automated-cherry-pick-of-#1206-upstream-release-1.0

Creating pull request for carlory:automated-cherry-pick-of-#1206-upstream-release-1.0 into release-1.0 in karmada-io/karmada

https://github.com/karmada-io/karmada/pull/1217

+++ Returning you to the script-cherry-pick-pull branch and cleaning up.
```

**Comment by carlory** [comment]
#1217 is created by the `hack/cherry_pick_pull.sh` script

File: hack/cherry_pick_pull.sh [hunk 1]
```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env bash
+
```
Comment (RainbowMango) [inline hack/cherry_pick_pull.sh:2]: Please keep the Kubernetes header here.

File: docs/userguide/cherry-picks.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
+# Overview
+
+This document explains how cherry picks are managed on release branches within
+the `karmada-io/karmada` repository.
+A common use case for this task is backporting PRs from master to release
+branches.
+
```
Comment (RainbowMango) [inline docs/userguide/cherry-picks.md:7]:
```diff
diff --git a/docs/userguide/cherry-picks.md b/docs/userguide/cherry-picks.md
index 3bb07ebf..1d2e6ee4 100644
--- a/docs/userguide/cherry-picks.md
+++ b/docs/userguide/cherry-picks.md
@@ -5,6 +5,8 @@ the `karmada-io/karmada` repository.
 A common use case for this task is backporting PRs from master to release
 branches.

+> This doc is lifted from [Kubernetes cherry-pick](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md).
+
 - [Prerequisites](#prerequisites)
 - [What Kind of PRs are Good for Cherry Picks](#what-kind-of-prs-are-good-for-cherry-picks)
 - [Initiate a Cherry Pick](#initiate-a-cherry-pick)

```

**Review by RainbowMango — COMMENTED** [review]
Thanks @carlory .

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**